python/uv/private/BUILD.bazel

```diff
@@ -88,3 +98,15 @@ bzl_library(
     srcs = ["versions.bzl"],
     visibility = ["//python/uv:__subpackages__"],
 )
+
+py_binary(
+    name = "pip_compile",
+    srcs = ["pip_compile.py"],
+    data = [
+        "//python/uv:current_toolchain",
```

Comment:
Here we use a `py_binary` to depend on 2 toolchains at once. The `uv:current_toolchain` should be always transitioned to the host platform, just like the python toolchain. We could in theory just have a `pip_compile.py` script that we launch like:
```
$(PYTHON3) $(rootpath pip_compile.py) $(rootpath params.file)
```

Where we add the `uv:current_toolchain` into params.